lib/python/src/bailo/helper/mirroredModel.py

```diff
@@ -0,0 +1,281 @@
+from __future__ import annotations
+
+import logging
+import os
+import shutil
+import tempfile
+import warnings
+from typing import Any
+
+from semantic_version import Version
+
+# isort: split
+
+from bailo.core.client import Client
+from bailo.core.enums import CollaboratorEntry, EntryKind, MinimalSchema, ModelVisibility
+from bailo.core.exceptions import BailoException
+from bailo.core.utils import NestedDict
+from bailo.helper.entry import Entry
+from bailo.helper.release import Release
+
+logger = logging.getLogger(__name__)
+
+
+class MirroredModel(Entry):
+    """Represent a mirrored model within Bailo.
+
+    :param client: A client object used to interact with Bailo
+    :param model_id: A unique ID for the mirrored model
+    :param name: Name of mirrored model
+    :param description: Description of mirrored model
+    :param organisation: Organisation responsible for the mirrored model, defaults to None
+    :param state: Development readiness of the model, defaults to None
+    :param collaborators: list of CollaboratorEntry to define who the mirrored model's collaborators (a.k.a. mirrored model access) are, defaults to None
+    :param visibility: Visibility of model, using ModelVisibility enum (e.g Public or Private), defaults to None
+    """
+
+    def __init__(
+        self,
+        client: Client,
+        model_id: str,
+        name: str,
+        description: str,
+        sourceModelId: str,
+        organisation: str | None = None,
+        state: str | None = None,
+        collaborators: list[CollaboratorEntry] | None = None,
+        visibility: ModelVisibility | None = None,
+    ) -> None:
+        super().__init__(
+            client=client,
+            id=model_id,
+            name=name,
+            description=description,
+            kind=EntryKind.MIRRORED_MODEL,
+            visibility=visibility,
+            organisation=organisation,
+            state=state,
+            collaborators=collaborators,
+        )
+        self.sourceModelId = sourceModelId
+        self.model_id = model_id
+
+    @classmethod
+    def create(
+        cls,
+        client: Client,
+        name: str,
+        description: str,
+        sourceModelId: str,
+        organisation: str | None = None,
+        state: str | None = None,
+        collaborators: list[CollaboratorEntry] | None = None,
+        visibility: ModelVisibility | None = None
+    ) -> MirroredModel:
+        """Build a mirrored model from Bailo and upload it.
+
+        :param client: A client object used to interact with Bailo
+        :param name: Name of model
+        :param description: Description of model
+        :param organisation: Organisation responsible for the model, defaults to None
+        :param state: Development readiness of the model, defaults to None
```

Comment:
```suggestion
        :param state: Development readiness of the mirrored model, defaults to None
```
Plenty more like this - use the RegEx in https://github.com/gchq/Bailo/pull/2722/files#r2468498748 to find these.